plugins/outputs/sumologic/README.md

```diff
@@ -0,0 +1,68 @@
+# SumoLogic Output Plugin
+
+This plugin sends metrics to [Sumologic HTTP Source](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source)
+in HTTP messages, encoded using one of the output data formats.
+Currently metrics can be sent using one of the following data formats, supported
+by Sumologic HTTP Source:
+
+  * `graphite` - for Content-Type of `application/vnd.sumologic.graphite`
+  * `carbon2` - for Content-Type of `application/vnd.sumologic.carbon2`
+  * `prometheus` - for Content-Type of `application/vnd.sumologic.prometheus`
+
+### Configuration:
+
+```toml
+# A plugin that can send metrics to Sumo Logic HTTP metric collector.
+[[outputs.sumologic]]
+  ## Unique URL generated for your HTTP Metrics Source.
+  ## This is the address to send metrics to.
+  # url = "https://events.sumologic.net/receiver/v1/http/<UniqueHTTPCollectorCode>"
+
+  ## Data format to be used for sending metrics.
+  ## This will set the "Content-Type" header accordingly.
+  ## Currently supported formats: 
+  ## * graphite - for Content-Type of application/vnd.sumologic.graphite
+  ## * carbon2 - for Content-Type of application/vnd.sumologic.carbon2
+  ## * prometheus - for Content-Type of application/vnd.sumologic.prometheus
+  ##
+  ## More information can be found at:
+  ## https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#content-type-headers-for-metrics
+  ##
+  ## NOTE:
+  ## When unset, telegraf will by default use the influx serializer which is currently unsupported
+  ## in HTTP Source.
+  data_format = "carbon2"
+
+  ## Timeout used for HTTP request
+  # timeout = "5s"
+  
+  ## HTTP method, one of: "POST" or "PUT". "POST" is used by default if unset.
+  # method = "POST"
+
+  ## HTTP Content-Encoding for writing request body. Can be set to "gzip" to compress the body.
+  ## This will set the "Content-Type" header accordingly.
+  # content_encoding = "gzip"
+
+  ## Max HTTP request body size in bytes before compression (if applied).
+  ## By default 1MB is recommended.
+  ## NOTE:
+  ## Bear in mind that in some serializer a metric even though serialized to multiple
+  ## lines cannot be split any further so setting this very low might not work
+  ## as expected.
+  # max_request_body_size = 1_000_000
+
+  ## Additional, sumologic specific HTTP header that can be set.
+  ## Full list can be found here:
+  ## https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#supported-http-headers
+  [outputs.sumologic.headers]
```

Comment:
Rather than expose the header, can we provide a wrapper around this to make the metadata properties first class citizens?  eg a property called sourceCategory as opposed to a header property?

Reply:
Sure. To be precise: you're suggesting to drop exposing the headers altogether and then only expose all of the 

* `X-Sumo-Name`
* `X-Sumo-Host`
* `X-Sumo-Category`
* `X-Sumo-Dimensions`

?

Let's say calling them: `x_sumo_name`, `x_sumo_host` etc (to stick with the snake case as in other config variables)?

Reply:
I'd even go further and called them `sumo_name`, `sumo_host` without `x` prefix.

But I'm not sure we should use the headers names in here and not the description names instead as that's how it's used across Sumo.

So instead of exposing `sumo_category` I'd rather expose it as `source_category`.
Same with `source_host`, `source_name` etc.

Reply:
@frankreno what do you think?

Reply:
@perk-sumo @frankreno 

I've got a patch prepared with the following (obviously easily swappable for e.g. `source_name`, `source_host` etc).

At the same time dropping the exposed headers field: @frankreno please confirm that that there are no viable use cases for our customers adding custom headers when sending data to http source.

```
  ## Desired source name. 
  ## Useful if you want to override the source name configured for the source.
  # sumo_name = ""

  ## Desired host name.
  ## Useful if you want to override the source host configured for the source.
  # sumo_host = ""

  ## Desired source category.
  ## Useful if you want to override the source category configured for the source.
  # sumo_category = ""

  ## Comma-separated key=value list of dimensions to apply to every metric.
  ## Custom dimensions will allow you to query your metrics at a more granular level.
  # sumo_dimensions = ""
```

Let me know

Reply:
i think sumo prefix is not necessary, since it is the Sumo Logic output plugin.  I think we should be consistent with other projects:

https://github.com/SumoLogic/fluentd-output-sumologic#configuration

https://github.com/SumoLogic/logstash-output-sumologic#parameters-of-plugin

I think `source_name`, `source_category`, `source_host`, `dimensions`